src/sqlancer/yugabyte/ysql/gen/YSQLCommon.java

```diff
@@ -159,8 +159,14 @@ public static void generateWith(StringBuilder sb, YSQLGlobalState globalState, E
             }
         } else if (Randomly.getBoolean()) {
             errors.add("Cannot use TABLEGROUP with TEMP table");
-            sb.append(" TABLEGROUP tg").append(
-                    Randomly.getNotCachedInteger(1, (int) YSQLTableGroupGenerator.UNIQUE_TABLEGROUP_COUNTER.get()));
+            try {
```

Comment:
Why we need this try catch?

Reply:
I think a cleaner way would be to let `getDatabaseIsColocated` return `Optional` (in case an empty result set might be expected for the query that `getDatabaseIsColocated` executes).

Reply:
What does it mean by saying return `optional`. Could give a example?